onnxruntime/core/providers/cpu/math/element_wise_ops.h

```diff
@@ -648,11 +762,74 @@ void BroadcastLoopSpan(TBroadcaster& bc, Output& output, Input0Scalar input0scal
+  } else if (input1_size == 1) {
+    concurrency::ThreadPool::TryParallelFor(tp, output_size,
+                                {static_cast<float>(sizeof(TInput)), static_cast<float>(sizeof(TOutput)), unit_cost},
+                               [=](std::ptrdiff_t first, std::ptrdiff_t last) {
+                                 size_t count = static_cast<size_t>(last - first);
+                                 EigenVectorMap<TOutput> output_map(output_ptr + first, count);
+                                 ConstEigenVectorMap<TInput> input0_map(input0_ptr + first, count);
+                                 input1scalar(output_map, input0_map, *input1_ptr);
+                               });
+  } else {
+    concurrency::ThreadPool::TryParallelFor(tp, output_size,
+                                {static_cast<float>(sizeof(TInput)), static_cast<float>(sizeof(TOutput)), unit_cost},
+                               [=](std::ptrdiff_t first, std::ptrdiff_t last) {
+                                 size_t count = static_cast<size_t>(last - first);
+                                 EigenVectorMap<TOutput> output_map(output_ptr + first, count);
+                                 ConstEigenVectorMap<TInput> input0_map(input0_ptr + first, count);
+                                 ConstEigenVectorMap<TInput> input1_map(input1_ptr + first, count); 
+                                 general(output_map, input0_map, input1_map);
+                               });
+  }
+}
 
+template <typename TInput, typename TOutput, typename Input0Scalar, typename Input1Scalar, typename General>
+Status BroadcastTwo(OpKernelContext& context, Input0Scalar input0scalar, Input1Scalar input1scalar, General general, double unit_cost=-1.0f) {
+  if (unit_cost == -1.0f) { // no paralellization 
+    TBroadcaster<TInput, TInput> bc(*context.Input<Tensor>(0), *context.Input<Tensor>(1));
+    TBroadcastOutput<TOutput> output(bc.GetSpanSize(), *context.Output(0, bc.GetOutputShape()));
+    BroadcastLoop(bc, output, input0scalar, input1scalar, general);
+  } else {
+  const Tensor* input0_tensor = context.Input<Tensor>(0);
+  const Tensor* input1_tensor = context.Input<Tensor>(1);
+  TBroadcaster<TInput, TInput> bc(*input0_tensor, *input1_tensor);
+  Tensor& output_tensor = *context.Output(0, bc.GetOutputShape());
+  auto span_size = bc.GetSpanSize();
+  int64_t output_size = output_tensor.Shape().Size();
+
+  concurrency::ThreadPool* tp = context.GetOperatorThreadPool();
+  if (span_size != 0) {
+    if (output_size == static_cast<int64_t>(span_size)) {  // Only one big span for all data, parallel inside it
+      BroadcastOneSpan(tp, unit_cost, output_tensor.MutableData<TOutput>(), output_size,
+                       input0_tensor->Data<TInput>(), input0_tensor->Shape().Size(),
+                       input1_tensor->Data<TInput>(), input1_tensor->Shape().Size(),
+                       input0scalar, input1scalar, general);
+    } else {
+      concurrency::ThreadPool::TryParallelFor(
```

Comment:
Would it be correct to assert ((output_size % span_size)==0), or can the uneven case occur?

Reply:
yes, added.